### PR TITLE
enable maligned lint check

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- enable maligned lint check
- perform code fixes

### CODE REVIEW

File: testdata/project-v2/Makefile [hunk 1]
```diff
@@ -52,7 +52,7 @@ vet:
 
 # Generate code
 generate: controller-gen
-	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths="./..."
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
```
Comment (Adirio) [inline testdata/project-v2/Makefile]: Fixed and tracked by #1288

File: cmd/init_project.go [hunk 2]
```diff
@@ -68,21 +68,23 @@ kubebuilder init --domain example.org --license apache2 --owner "The Kubernetes
...
 	depArgs []string
 
-	// final result
-	scaffolder scaffold.ProjectScaffolder
+	// flags
+	fetchDeps          bool
+	skipGoVersionCheck bool
+
+	// @Deprecated
+	dep bool
```
Comment (Adirio) [inline cmd/init_project.go]: Why split this from the other deprecated flags?

Comment (camilamacedo86) [reply cmd/init_project.go]: In order to solve the `maligned` issue check, I need to to add the bools at the end. 
This is about memory space usage.

**Comment by Adirio** [comment]
LGTM from a code perspective.

The benefit from adding maligned check to kubebuilder isn't that big as most of the types, if not all of them, will only have 1 or 2 instances, so the memory reduction will be in the order of bytes. Even if small, it is still a benefit and if it doens't introduce any harm it is an enhancement.

File: cmd/init_project.go [hunk 3]
```diff
@@ -68,21 +68,23 @@ kubebuilder init --domain example.org --license apache2 --owner "The Kubernetes
...
+	// @Deprecated
 	depArgs []string
 
-	// final result
-	scaffolder scaffold.ProjectScaffolder
+	// flags
+	fetchDeps          bool
+	skipGoVersionCheck bool
+
+	// @Deprecated
```
Comment (mengqiy) [inline cmd/init_project.go]: > @Deprecated

Is it required to use this format?
Can you please point me to the document if any?

Comment (camilamacedo86) [reply cmd/init_project.go]: No. The check has no relation with the comment.
I will replace for the previous comment since shows that you prefer.

Comment (camilamacedo86) [reply cmd/init_project.go]: Done.

**Comment by camilamacedo86** [comment]
@mengqiy could you please re-apply the /lgtm label here?

**Comment by mengqiy** [comment]
@camilamacedo86 It needs rebase

**Comment by camilamacedo86** [comment]
Hi @mengqiy tks for the review. 
Rebased just missing 
/lgtm

**Comment by Adirio** [comment]
@camilamacedo86 Travis should just require to be re-triggered.

**Comment by mengqiy** [comment]
Need rebase again...

**Pull request merged**